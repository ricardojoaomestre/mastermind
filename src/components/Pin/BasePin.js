import styled from 'styled-components';

const BasePin = styled.div`
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: ${props => props.color || "#F0F1F6"};

    &:hover {
        cursor: pointer;
    }
`;

export default BasePin;
