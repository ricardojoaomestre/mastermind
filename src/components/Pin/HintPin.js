import styled from 'styled-components';
import BasePin from './BasePin';

const StyledHintPin = styled(BasePin)`
    width: 14px;
    height: 14px;
    border-radius: 8px;
    border: 1px solid #C4C4C4;
    background-color: ${props => props.value === "correct" ? "#000000" : props.value === "semicorrect" ? "#D9DADC" : "transparent"};

    &:hover {
        cursor: default;
    }
`;

const HintPin =  (props) => <StyledHintPin {...props}/>;

export default HintPin;