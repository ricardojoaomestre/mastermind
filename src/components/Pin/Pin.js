import styled from 'styled-components';
import BasePin from './BasePin';

const StyledPin = styled(BasePin)`
    width: 42px;
    height: 42px;
    border-radius: 21px;
`;

const Pin =  (props) => <StyledPin {...props} />;

export default Pin;