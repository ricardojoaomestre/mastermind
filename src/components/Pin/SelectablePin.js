import styled from 'styled-components';
import Pin from './Pin';
import { IoCheckmarkOutline } from 'react-icons/io5';

const PinIcon = styled(IoCheckmarkOutline)`
    width: 20px;
    height: 20px;
    stroke: #FFFFFF;
`;

const SelectablePin =  ({selected, ...otherProps}) => (
    <Pin {...otherProps} >
        {selected && <PinIcon />}
    </Pin>
);

export default SelectablePin;