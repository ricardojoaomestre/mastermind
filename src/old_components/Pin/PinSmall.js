import styled from 'styled-components';
import BasePin from './BasePin';

const StyledPinSmall = styled(BasePin)`
    width: 32px;
    height: 32px;
    border-radius: 16px;
`;

const PinSmall =  (props) => <StyledPinSmall {...props} />;

export default PinSmall;