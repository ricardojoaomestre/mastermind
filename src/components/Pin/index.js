import Pin from "./Pin";
import PinSmall from "./PinSmall";
import HintPin from "./HintPin";
import SelectablePin from "./SelectablePin";

export { Pin, PinSmall, HintPin, SelectablePin };
