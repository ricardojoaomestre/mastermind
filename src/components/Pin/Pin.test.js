import { render, cleanup } from '@testing-library/react';

import Pin from './Pin';

afterEach(cleanup);

describe('Pin', () => {
    it("must render as expected", () => {
        const { container } = render(<Pin />);
        expect(container.firstChild).toMatchSnapshot();
    });
});