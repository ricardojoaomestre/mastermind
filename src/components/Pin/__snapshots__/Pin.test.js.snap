// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin must render as expected 1`] = `
<div
  class="sc-bdfBQB sc-gsTEea ioqCIK dCYNna"
/>
`;
